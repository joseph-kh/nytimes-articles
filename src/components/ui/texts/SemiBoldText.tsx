import React from "react";
import { StyleSheet, Text as TextElement, TextProps } from "react-native";
import { moderateScale } from "@/utils/scaler";

type Props = {
  fontSize?: number;
};

const SemiBoldText: React.FC<Props & TextProps> = (props) => {
  const { style, fontSize, children } = props;

  return (
    <TextElement
      {...props}
      style={StyleSheet.flatten([
        styles.textStyle,
        style,
        !!fontSize && { fontSize: moderateScale(fontSize) },
      ])}
    >
      {children}
    </TextElement>
  );
};

export default SemiBoldText;

const styles = StyleSheet.create({
  textStyle: {
    fontFamily: "SemiBold",
    fontSize: moderateScale(14),
  },
});
