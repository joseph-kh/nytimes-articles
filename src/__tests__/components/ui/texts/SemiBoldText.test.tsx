import React from "react";
import renderer from "react-test-renderer";
import SemiBoldText from "@/components/ui/texts/SemiBoldText";

describe("SemiBoldText Component", () => {
  it("renders correctly with default props", () => {
    const tree = renderer
      .create(<SemiBoldText>Hello, World!</SemiBoldText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom font size", () => {
    const tree = renderer
      .create(<SemiBoldText fontSize={18}>Custom Font Size</SemiBoldText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
