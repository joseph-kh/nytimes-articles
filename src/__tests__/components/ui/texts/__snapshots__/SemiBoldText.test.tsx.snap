// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemiBoldText Component renders correctly with custom font size 1`] = `
<Text
  fontSize={18}
  style={
    {
      "fontFamily": "SemiBold",
      "fontSize": 27,
    }
  }
>
  Custom Font Size
</Text>
`;

exports[`SemiBoldText Component renders correctly with default props 1`] = `
<Text
  style={
    {
      "fontFamily": "SemiBold",
      "fontSize": 21,
    }
  }
>
  Hello, World!
</Text>
`;
